src/modeler/AutoRest.Swagger/Model/ValidationUtilities.cs

```diff
@@ -14,6 +14,7 @@ public static class ValidationUtilities
     {
         private static readonly string XmsPageable = "x-ms-pageable";
         private static readonly Regex TrackedResRegEx = new Regex(@".+/Resource$", RegexOptions.IgnoreCase);
+        private static readonly Regex resourceProviderPathPattern = new Regex(@"/providers/(?<resPath>[^{/]+)/", RegexOptions.IgnoreCase);
```

Comment:
Per offline talk with deepak, I've updated rule to check only for one RP in case of multiple RPs found in swagger.